### PR TITLE
Add docstrings for exegesis python bindings

### PR DESCRIPTION
**Pull request by boomanaiden154**
This patch adds docstrings for the ExegesisBenchmark python bindings so that they are better documented when working on them and also better documented when using them.

### CODE REVIEW

File: gematria/datasets/python/exegesis_benchmark.cc [hunk 1]
```diff
@@ -67,22 +68,73 @@ PYBIND11_MODULE(exegesis_benchmark, m) {
...
+      .def(
+          "create",
+          []() -> absl::StatusOr<std::unique_ptr<ExegesisBenchmark>> {
+            InitializeForExegesisOnce();
+
+            return LlvmExpectedToStatusOr(ExegesisBenchmark::create());
+          },
+          R"(Creates a ExegesisBenchmark Instance.
+
+          Does the necessary initialization to run Exegesis and then creates
```
Comment (orodley) [inline gematria/datasets/python/exegesis_benchmark.cc:80]: What happens if we call this twice -- does it do this initialisation again, or only the first time it's called?

File: gematria/datasets/python/exegesis_benchmark.cc [hunk 2]
```diff
@@ -67,22 +68,73 @@ PYBIND11_MODULE(exegesis_benchmark, m) {
...
+
+          Takes a BenchmarkCode instance and then executes it, collecting
+          performance information at the same time.
+
+          Args:
+            input_benchmark: The BenchmarkCode instance formed from the block and
+              annotations of interest that should be benchmarked.
+          
+          Returns:
+            A floating point value represnting the inverse throughput of the block.
```
Comment (orodley) [inline gematria/datasets/python/exegesis_benchmark.cc:133]: A bit more information on the units (IIUC this is a count of cycles taken to execute the block) would be useful here for interpreting the result

Comment (orodley) [inline gematria/datasets/python/exegesis_benchmark.cc:133]: Typo: represnting

File: gematria/datasets/python/exegesis_benchmark.cc [hunk 3]
```diff
@@ -67,22 +68,73 @@ PYBIND11_MODULE(exegesis_benchmark, m) {
...
+          "create",
+          []() -> absl::StatusOr<std::unique_ptr<ExegesisBenchmark>> {
+            InitializeForExegesisOnce();
+
+            return LlvmExpectedToStatusOr(ExegesisBenchmark::create());
+          },
+          R"(Creates a ExegesisBenchmark Instance.
+
+          Does the necessary initialization to run Exegesis and then creates
+          a ExegesisBenchmark that can then be used to execute annotated
```
Comment (orodley) [inline gematria/datasets/python/exegesis_benchmark.cc:81]: Nit: a -> an (and below in the "Returns" section)

File: gematria/datasets/python/exegesis_benchmark.cc (hunk 1 above)
Comment (orodley) [inline gematria/datasets/python/exegesis_benchmark.cc:80]: Nit: I'd drop both the "then"s in this paragraph, as they're not necessary.

**Review by orodley — APPROVED** [review]

File: gematria/datasets/python/exegesis_benchmark.cc (hunk 1 above)
Comment (boomanaiden154) [reply gematria/datasets/python/exegesis_benchmark.cc:80]: Clarified.

File: gematria/datasets/python/exegesis_benchmark.cc [hunk 4]
```diff
@@ -67,22 +68,76 @@ PYBIND11_MODULE(exegesis_benchmark, m) {
...
-        return LlvmExpectedToStatusOr(Self.benchmarkBasicBlock(InputBenchmark));
-      });
+      .def(
+          "create",
+          []() -> absl::StatusOr<std::unique_ptr<ExegesisBenchmark>> {
+            InitializeForExegesisOnce();
+
+            return LlvmExpectedToStatusOr(ExegesisBenchmark::create());
+          },
+          R"(Creates a ExegesisBenchmark Instance.
```
Comment (ondrasej) [inline gematria/datasets/python/exegesis_benchmark.cc:78]: s/a/an/

File: gematria/datasets/python/exegesis_benchmark.cc [hunk 5]
```diff
@@ -67,22 +68,76 @@ PYBIND11_MODULE(exegesis_benchmark, m) {
...
+
+          Does the necessary initialization to run Exegesis and creates a
```
Comment (ondrasej) [inline gematria/datasets/python/exegesis_benchmark.cc:80]: ditto

**Review by ondrasej — APPROVED** [review]

**Pull request merged**